storage/localstore/mode_set.go

```diff
@@ -201,48 +202,74 @@ func (db *DB) setSync(batch *leveldb.Batch, addr chunk.Address, mode chunk.ModeS
 	item.StoreTimestamp = i.StoreTimestamp
 	item.BinID = i.BinID
 
-	// moveToGc toggles the deletion of the item from pushsync index
-	// it will be false in the case pull sync was called but push sync was meant on that chunk (!tag.Anonymous)
-	moveToGc := true
-	if db.tags != nil {
-		i, err = db.pushIndex.Get(item)
-		switch err {
-		case nil:
-			tag, _ := db.tags.Get(i.Tag)
-			if tag != nil {
-				if tag.Anonymous {
-					// anonymous - only pull sync
-					switch mode {
-					case chunk.ModeSetSyncPull:
-						// this will not get called twice because we remove the item once after the !moveToGc check
-						tag.Inc(chunk.StateSent)
-						// this is needed since pushsync is checking if `tag.Done(chunk.StateSynced)` and when overlapping
-						// chunks are synced by both push and pull sync we have a problem. as an interim solution we increment this too
-						tag.Inc(chunk.StateSynced)
-					case chunk.ModeSetSyncPush:
-						// do nothing - this should not be possible. just log and return
-						log.Warn("chunk marked as push-synced but should be anonymous!", "tag.Uid", tag.Uid)
-						moveToGc = false
-					}
-				} else {
-					// not anonymous - push and pull should be used
-					switch mode {
-					case chunk.ModeSetSyncPull:
-						moveToGc = false
-					case chunk.ModeSetSyncPush:
-						tag.Inc(chunk.StateSynced)
-					}
-				}
+	switch mode {
```

Comment:
It would be good to update setSync comment with some notion on anonymous tags, so that the exported api can be documented.

Reply:
added comment to the enclosing function. let me know if it's enough